### PR TITLE
Remove use of colors

### PR DESCRIPTION
**Pull request by marbar3778**
- Colors.go was removed from tendermint

Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

### CODE REVIEW

**Comment by melekes** [comment]
Are you sure we don't need it? Maybe move them from tendermint-core instead of removing?

**Comment by marbar3778** [comment]
Is there a benefit to having colors in the terminal? If so, I can add that section into here.

**Comment by alessio** [comment]
There should be a output formatting function that may conditionally enable colours on screen IMHO

File: Makefile [hunk 1]
```diff
@@ -2,10 +2,18 @@ GOTOOLS := github.com/golangci/golangci-lint/cmd/golangci-lint
 
 PDFFLAGS := -pdf --nodefraction=0.1
 
+LINKER_FLAG := -ldflags "-X main.COLOR_ON=on"
```
Comment (alessio) [inline Makefile]:
```suggestion
LDFLAGS := -ldflags "-X main.COLOR_ON=on"
```

File: Makefile [hunk 2]
```diff
@@ -2,10 +2,18 @@ GOTOOLS := github.com/golangci/golangci-lint/cmd/golangci-lint
...
+
 all: lint test install
 
+
+
 install:
+ifeq ($(COLORS_ON),)
 	go install ./cmd/iaviewer
+else
+	go install $(LINKER_FLAG) ./cmd/iaviewer
```
Comment (alessio) [inline Makefile]:
```suggestion
	go install $(LDFLAGS) ./cmd/iaviewer
```

File: util.go [hunk 3]
```diff
@@ -102,3 +104,61 @@ func sortByteSlices(src [][]byte) [][]byte {
...
+func Blue(args ...interface{}) string {
+	return treatAll(ANSIFgBlue, args...)
+}
+
+func Cyan(args ...interface{}) string {
+	return treatAll(ANSIFgCyan, args...)
+}
+
+func ColoredBytes(data []byte, textColor, bytesColor func(...interface{}) string) string {
+	colors := os.Getenv("COLORS_ON")
```
Comment (alessio) [inline util.go]: Mmmmh... If we really want go for an environment variable let's at least prefix it with `IAVL_`:

```suggestion
	colors := os.Getenv("IAVL_COLORS_ON")
```

File: util.go [hunk 4]
```diff
@@ -102,3 +104,61 @@ func sortByteSlices(src [][]byte) [][]byte {
...
+
+func Blue(args ...interface{}) string {
+	return treatAll(ANSIFgBlue, args...)
+}
+
+func Cyan(args ...interface{}) string {
+	return treatAll(ANSIFgCyan, args...)
+}
+
+func ColoredBytes(data []byte, textColor, bytesColor func(...interface{}) string) string {
```
Comment (alessio) [inline util.go]: Please add a godoc comment to explain what this function does, plus document the environment variable usage.

File: util.go [hunk 5]
```diff
@@ -102,3 +104,64 @@ func sortByteSlices(src [][]byte) [][]byte {
...
+
+func Blue(args ...interface{}) string {
+	return treatAll(ANSIFgBlue, args...)
+}
+
+func Cyan(args ...interface{}) string {
+	return treatAll(ANSIFgCyan, args...)
+}
+
+// Coloredbytes takes in the byte that you would like to show as a string and byte
```
Comment (alessio) [inline util.go]:
```suggestion
// ColoredBytes takes in the byte that you would like to show as a string and byte
```

File: util.go [hunk 6]
```diff
@@ -102,3 +104,64 @@ func sortByteSlices(src [][]byte) [][]byte {
...
+// and display them in different colors, if `TENDERMINT_IAVL_COLORS_ON` is set to on.
```
Comment (alessio) [inline util.go]:
```suggestion
// and display them in a human readable format.
// If the environment variable `TENDERMINT_IAVL_COLORS_ON` is set to a non empty string,
// strings and bytes will be displayed with different colours. 
```

File: util.go [hunk 7]
```diff
@@ -102,3 +104,64 @@ func sortByteSlices(src [][]byte) [][]byte {
...
+}
+
+func Cyan(args ...interface{}) string {
+	return treatAll(ANSIFgCyan, args...)
+}
+
+// ColoredBytes takes in the byte that you would like to show as a string and byte
+// and will display them in different colors for readability, if `TENDERMINT_IAVL_COLORS_ON` is set to on.
+func ColoredBytes(data []byte, textColor, bytesColor func(...interface{}) string) string {
+	// If TENDERMINT_IAVL_COLORS_ON is set to on then colors will be used for bytes and strings
```
Comment (alessio) [inline util.go]: This is wrong. To enable colours `TENDERMINT_IAVL_COLORS_ON` needs just to be a non-empty string

File: util.go [hunk 8]
```diff
@@ -102,3 +104,64 @@ func sortByteSlices(src [][]byte) [][]byte {
...
+}
+
+func Cyan(args ...interface{}) string {
+	return treatAll(ANSIFgCyan, args...)
+}
+
+// ColoredBytes takes in the byte that you would like to show as a string and byte
+// and will display them in a human readable format.
+func ColoredBytes(data []byte, textColor, bytesColor func(...interface{}) string) string {
+	// If TENDERMINT_IAVL_COLORS_ON is set to a non-empty string then colors will be used for bytes and strings
```
Comment (alessio) [inline util.go]:
```suggestion
	// If the environment variable TENDERMINT_IAVL_COLORS_ON is set to a non-empty string then different colors will be used for bytes and strings.
```

Comment (alessio) [reply util.go]: Plus, please incorporate this in the godoc comment. Then we are done :)

**Pull request merged**